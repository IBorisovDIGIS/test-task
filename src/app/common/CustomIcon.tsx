import { Icon, makeStyles } from '@material-ui/core';
import React from 'react';
import clsx from 'clsx';

const useStyles = makeStyles({
  imageIcon: {
    height: 24,
    width: 24,
  },
  iconRoot: {
    textAlign: 'center',
    height: '100%',
  },
});

interface ICustomIcon {
    icon: string;
    iconClasses?: string;
    alt?: string
}

const CustomIcon: React.FC<ICustomIcon> = ({ icon, iconClasses, ...props }: ICustomIcon) => {
  const classes = useStyles();

  return (
    <Icon classes={{ root: classes.iconRoot }}>
      <img
        className={clsx(classes.imageIcon, iconClasses)}
        src={icon}
        alt="alt"
        {...props}
      />
    </Icon>
  );
};

export default CustomIcon;

CustomIcon.defaultProps = {
  iconClasses: '',
  alt: '',
};
