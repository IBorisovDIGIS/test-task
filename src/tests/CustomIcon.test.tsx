// CustomIcon.tsx test
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import CustomIcon from '../app/common/CustomIcon';
import CopyIcon from '../assets/icons/copy.svg';

configure({ adapter: new Adapter() });

describe('Custom Icon should render correctly', () => {
  it('Should renders correctly', () => {
    const icon = shallow(<CustomIcon icon={CopyIcon} alt="icon" iconClasses="testClass" />);

    expect(toJson(icon)).toMatchSnapshot();
  });

  it('Should pass props correctly', () => {
    const icon = shallow(<CustomIcon icon={CopyIcon} alt="icon" iconClasses="testClass" />);

    expect(icon.find('img').prop('src')).toEqual(CopyIcon);
    expect(icon.find('img').prop('alt')).toEqual('icon');
    expect(icon.find('img').prop('className')).toContain('testClass');
  });
});
