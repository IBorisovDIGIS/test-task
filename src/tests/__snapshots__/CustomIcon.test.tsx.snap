// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Icon should render correctly Should renders correctly 1`] = `
<WithStyles(ForwardRef(Icon))
  classes={
    Object {
      "root": "makeStyles-iconRoot-2",
    }
  }
>
  <img
    alt="icon"
    className="makeStyles-imageIcon-1 testClass"
    src="copy.svg"
  />
</WithStyles(ForwardRef(Icon))>
`;
